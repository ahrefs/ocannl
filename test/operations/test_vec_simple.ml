open Base
open Ocannl.Operation.DSL_modules

let () =
  let module Backend = (val Backends.fresh_backend ()) in
  (* Generate random bits *)
  let%op random_bits = threefry4x32 (embed_self_id ()) 42L in

  (* Convert to uniform floats *)
  let%op uniform_floats = uint4x32_to_prec_uniform random_bits in
  Ir.Tnode.update_prec uniform_floats.value Ir.Ops.single;

  (* Compile and run *)
  Ocannl.Train.set_hosted uniform_floats.value;
  ignore (Ocannl.Train.forward_once (module Backend) uniform_floats);
  let result = Ir.Tnode.get_values uniform_floats.value in

  (* Print results *)
  Stdio.printf "Generated %d uniform random numbers:\n" (Array.length result);
  Array.iteri result ~f:(fun i x -> Stdio.printf "  [%d]: %.4g\n" i x)
