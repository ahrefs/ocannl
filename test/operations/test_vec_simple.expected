Retrieving commandline, environment, or config file variable ocannl_log_level
Found 0, in the config file
Generated 4 uniform random numbers:
  [0]: 0.440217
  [1]: 0.459163
  [2]: 0.927131
  [3]: 0.356514
